bundles/org.openhab.binding.gree/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -34,6 +35,11 @@
 				<unitLabel>seconds</unitLabel>
 				<advanced>true</advanced>
 			</parameter>
+			<parameter name="currentTemperatureOffset" type="decimal" step="0.5" required="true" unit="C">
+				<default>-40</default>
+				<unitLabel>Degrees Celsius</unitLabel>
+				<advanced>true</advanced>
+			</parameter>
```

Comment:
I don't think it's intuitive if configuring an offset of 0, would result in an offset of +40°C. Can you subtract the internal offset in the code, so an offset of 0 is really 0?

Reply:
I think the idea here is that the device returns the data with a 40 degree offset. But that value 40 is not guaranteed. So therefor this parameter is there correct that. If it would always be 40 this offset would not be needed or maybe could be a boolean. To switch between 40 and 0 in case there is no offset. So it very much depends if the 40 is fixed or not.

Reply:
Yes this was the exact reasoning behind it. On the Gree devices which I have installed at home, the offset is that of +40°C. So the offset required for calibration for this particular device is actually +40 and not 0.

Judging from online research this seems to be the case with the majority of devices but I cannot vouch for that since new models might behave differently.

The idea for this configuration parameter was that other devices might come with an offset which is different from +40. So the idea was that the binding would offset exactly by the amount which is specified in this parameter and not make any assumptions other than setting the default at +40 which seems to be the most common at the time.

If for some other Gree device, the offset is +30°C, the required value here would be +30 and not -10.

Reply:
@markus7017 you as the codeowner, WDYT? Are you aware of any device type that has a different offset than -40°C?

Reply:
Hmm, as you said it seems that all devices have an offset of 40°C. Having this offset configurable would allow the user to do adjustment when the temp is really 1°C off etc., which is a useful feature.
We have no proof that there are other offsets than 40°. Therefore I would make this assumption and do the math in the binding. In case a device has something != 40, this is adjustable as a work around. We could then decide how to model this (e.g,. by a different thing type), but so far I don't see this scenario.

Reply:
Maybe just remove this parameter and hardcode the offset for now. It's always possible to add it later, but once added it's less convenient to change it if another design would have been beter.

Reply:
I favor
- keeping the parameter, but as normalized offset
- hard code the -40 as constant in the code

So 0 means 0 for the user, but -40 for the binding, 1=-39, -2=-42 etc.

@taboneclayton it might me that 0.5 steps are supported

Reply:
@taboneclayton From my point of view this is the only open issue. Could you change it that way?

Reply:
Updated as suggested. I think this should address all requests for changes. However there is a merge conflict in `thing-types.xml` due to the introduction of spotless. Don't want to merge from upstream to avoid problems. Should I rebase to resolve the conflict?

Reply:
Yes do rebase and `git push --force-with-lease` back to GitHub.

Also do run `mvn spotless:apply` on your binding to make sure your changes also comply with the style, otherwise it might break the build of your binding.

Reply:
Done